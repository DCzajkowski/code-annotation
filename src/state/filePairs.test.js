import configureMockStore from 'redux-mock-store';
import thunk from 'redux-thunk';
import reducer, {
  initialState,
  LOAD,
  LOAD_SUCCESS,
  LOAD_FILE_PAIR,
  SET_FILE_PAIR,
  SET_CURRENT_PAIR,
  load,
  getCurrentFilePair,
} from './filePairs';

const mockStore = configureMockStore([thunk]);

describe('filePairs/reducer', () => {
  it('LOAD', () => {
    expect(reducer(initialState, { type: LOAD })).toMatchSnapshot();
  });

  it('LOAD_SUCCESS', () => {
    expect(
      reducer(
        { ...initialState, loading: true },
        {
          type: LOAD_SUCCESS,
          list: [1, 2, 3],
        }
      )
    ).toMatchSnapshot();
  });

  it('LOAD_FILE_PAIR', () => {
    expect(
      reducer(initialState, {
        type: LOAD_FILE_PAIR,
      })
    ).toMatchSnapshot();
  });

  it('SET_FILE_PAIR', () => {
    expect(
      reducer(initialState, {
        type: SET_FILE_PAIR,
        id: 1,
        diff: 'content',
        score: 0.987,
        leftBlobId: 'id1',
        rightBlobId: 'id2',
      })
    ).toMatchSnapshot();
  });

  it('SET_CURRENT_PAIR', () => {
    expect(
      reducer(initialState, {
        type: SET_CURRENT_PAIR,
        id: 1,
      })
    ).toMatchSnapshot();
  });
});

describe('filePairs/actions', () => {
  it('load', () => {
    const store = mockStore({
      filePairs: initialState,
    });
    const pairs = [{ id: 1 }, { id: 2 }];

    fetch.mockResponse(
      JSON.stringify({
        data: pairs,
      })
    );

    store.dispatch(load(1)).then(() => {
      expect(store.getActions()).toEqual([
        {
          type: LOAD,
        },
        {
          type: LOAD_SUCCESS,
          pairs,
        },
      ]);
    });
  });
});

describe('filePairs/selector', () => {
  it('getCurrentFilePair', () => {
    const state = {
      filePairs: {
        currentPairId: 1,
        pairs: { 1: 'pair' },
      },
    };
    expect(getCurrentFilePair(state)).toEqual('pair');
  });
});
