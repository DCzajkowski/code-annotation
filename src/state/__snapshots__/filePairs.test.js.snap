// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`filePairs/reducer LOAD 1`] = `
Object {
  "currentAnnotations": Object {},
  "currentPairId": null,
  "list": Array [],
  "loading": true,
  "pairs": Object {},
}
`;

exports[`filePairs/reducer LOAD_FILE_PAIR 1`] = `
Object {
  "currentAnnotations": Object {},
  "currentPairId": null,
  "list": Array [],
  "loading": true,
  "pairs": Object {},
}
`;

exports[`filePairs/reducer LOAD_SUCCESS 1`] = `
Object {
  "currentAnnotations": Object {},
  "currentPairId": null,
  "list": undefined,
  "loading": false,
  "pairs": Object {},
}
`;

exports[`filePairs/reducer SET_CURRENT_PAIR 1`] = `
Object {
  "currentAnnotations": Object {},
  "currentPairId": 1,
  "list": Array [],
  "loading": true,
  "pairs": Object {},
}
`;

exports[`filePairs/reducer SET_FILE_PAIR 1`] = `
Object {
  "currentAnnotations": Object {},
  "currentPairId": null,
  "list": Array [],
  "loading": false,
  "pairs": Object {
    "1": Object {
      "diff": "content",
      "leftBlobId": "id1",
      "rightBlobId": "id2",
      "score": 0.987,
    },
  },
}
`;
